src/Features/CSharp/Portable/Completion/CompletionProviders/NamedParameterCompletionProvider.cs

```diff
@@ -256,10 +256,9 @@ int IEqualityComparer<IParameterSymbol>.GetHashCode(IParameterSymbol obj)
         {
             return Task.FromResult<TextChange?>(new TextChange(
                 selectedItem.Span,
-                // Do not insert colon on <Tab> so that user can complete out a variable name that does not currently exist.
-                // ch == null is to support the old completion only.
-                // Do not insert an extra colon if colon has been explicitly typed.
-                (ch == null || ch == '\t' || ch == ':') ? selectedItem.DisplayText : selectedItem.GetEntireDisplayText()));
+                // Insert extra colon if commiting with '(' only: "method(parameter:(" is preferred to "method(parameter(".
```

Comment:
> Insert extra colon if commiting with '('

The `:` is never added by the completion item. The character typed which triggered the completion is passed through, allowing the `:` to be explicitly typed.

Reply:
@sharwell 
1. yes, it fixes #33667. 
2. Do you mean we should not special casing `(`?

Reply:
Correct, the display text here includes `:`, but the insertion text does not. We should not need to include any special behavior for the insertion itself.

Reply:
Sounds good. Thank you, @sharwell ! Let us try this way. Changes committed. Please re-review.